lib/bulk_insert/worker.rb

```diff
@@ -130,6 +139,14 @@ def insert_ignore
       end
     end
 
+    def primary_key_return_statement
+      if adapter_name =~ /\APostgreSQL/i
```

Comment:
The `on_conflict_statement` supports both `postgresql` and `postgis` as adapter names.  Perhaps line 143 should use the same logic?  postgis is a postgresql extension, and thus also supports `RETURNING`.

Reply:
+1